### ISSUE TITLE
Access is denied on shutdown

### ISSUE THREAD

**Issue by czgu4422** [basildane/WakeOnLAN#171]
After I install the WOL to PC-01, I can wake up my PC-02 from PC-01. But I cannot remotely shut PC-02 down. Err: Access is denied. But in PC-01 I created a bash script: "shutdown -s -m \\PC-02 -t 5", than run the bash script. I can shutdown PC-02 from PC-01. Why shutdown in WOL does not work?

**Comment by basildane** [comment]
Are your computers in a domain or a workgroup?  Workgroups can be more difficult to shutdown, and I have some options in the Shutdown panel for that, including the "Legacy" shutdown method.  [https://wol.aquilatech.com/help/shutdown-host/](https://wol.aquilatech.com/help/shutdown-host/)

**Comment by czgu4422** [comment]
Workgroup. Waking up and shutting down work fine if 2 computers are in same network (home). I can use this program to wake up home computer from my office, but cannot shut it down. Also remote desktop does not work. I think they have same problem. Because there is no place I can enter port number.

**Comment by basildane** [comment]
Are you using RDP on a port other than 3389?  You can setup a shortcut and it will use that.  Did you try legacy mode on shutdown?  Can you shutdown the remote computer manually - not using wol?

**Comment by czgu4422** [comment]
No, I use port 3389 for RDP (from Windows 10). Windows 10 RDP works fine (I’ve set port forwarding).
Yes, I use legacy mode. Both  shutdown and Remote desktop work fine if 2 PCs are in same network. But if 2 PCs are NOT in same network (most cases are one PC at home, and one in my office), Shutdown does not work.

The reason that I use Aquila software is that the software can wake up my PC. After I changed router, many WOL software stop working. First I thought that was the new router (Netgear RAX15) problem. But there was no router setting problem. So I started searching WOL software on line and tried one by one. Aquila software is only one I can find to wake up my PC from different network. Currently if I’m not at home and want to shutdown PC, I use windows 10 RDP, login to my home PC, then shut it down. So it’s OK the Shutdown feature does not work if PCs are not in the same network.

Thanks

czgu

**Comment by basildane** [comment]
OK, in order to shutdown a computer on a remote network, you need to have direct communications (windows ports) mapped to that remote network.  The normal way you would do that would be to use a VPN.  You didn't say, but if the remote machine is across the Internet, a VPN is a must.  If the machine is on your LAN but just on a different segment, then you must forward all the windows network ports to that network, this is usually the job of your local firewall.
